R/init.R

```diff
@@ -221,17 +221,30 @@ init <- function(data,
     }
   }
 
+  reporter <- teal.reporter::Reporter$new()$set_id(attr(filter, "app_id"))
+  if (is_arg_used(modules, "reporter") && length(extract_module(modules, "teal_module_previewer")) == 0) {
+    modules <- append_module(
+      modules,
+      reporter_previewer_module(server_args = list(previewer_buttons = c("download", "reset")))
+    )
+  }
+
   # Note regarding case `id = character(0)`:
   # rather than creating a submodule of this module, we directly modify
   # the UI and server with `id = character(0)` and calling the server function directly
   # Note: UI must be a function to support bookmarking.
   res <- list(
-    ui = function(request) ui_teal_with_splash(id = id, data = data, title = title, header = header, footer = footer),
+    ui = function(request) {
```

Comment:
Just curious, what's the purpose of the `request` argument?

Reply:
`request` argument is needed to set bookmarking 

<img width="859" alt="image" src="https://github.com/insightsengineering/teal/assets/6959016/bf4a8eb2-1d2c-4e58-b768-c84dddaf3159">